### ISSUE TITLE
XAML Islands apps need a way to supress "Caret Browsing" dialog

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/microsoft-ui-xaml#1873]
It might be totally impossible with our architecture.

Our hosting window doesn't even receive a _win32 window message_ it can suppress or transform; everything is going straight into the Xaml Island's input site.

/cc @Austin-Lamb who once had an idea about how we could solve this.

Austin, should I file this one on microsoft-ui-xaml for when WinUI 3.0 is available? :smile:

_Originally posted by @DHowett-MSFT in https://github.com/microsoft/terminal/issues/638#issuecomment-511255709_

**Comment by DHowett-MSFT** [comment]
More info from the downstream bug:

> This is actually almost impossible to fix! Digging through the code for the Xaml platform keyboard handler, it intercepts <kbd>F7</kbd> before we're allowed to handle it (and reject its propagation.) The official recommendation from that team is to catch it in our island host's `WM_KEYDOWN` handler and synthesize it ourselves.

**Comment by Austin-Lamb** [comment]
I'll see if I can test a workaround before I suggest it - there's a couple ideas I have but I'm not sure which, if any, may work.

**Comment by Austin-Lamb** [comment]
There is a way to disable this - before you call IDesktopWindowXamlSourceNative::PreTranslateMessage, you can check if the message you're about to give us is WM_KEYDOWN with wParam == VK_F7.  If so, don't give it to us.

@zadjii-msft - please reactivate if that doesn't work for you.

@marb2000 - we should find a way to add this as an option to some XAML Islands samples as we've had a couple customers request this.  But we shouldn't do it by default, as it does disable an accessibility feature, so just showing how to do it for customers that understand the impact would be better.

There's alternative mechanisms if using WPF as the hosting framework, I haven't tried WinForms yet.  So we could discuss the best way to add to the samples.

**Comment by DHowett-MSFT** [comment]
We're not actually calling PreTranslateMessage for any reason... we also still want to pass F7 through to the TerminalControl in the Xaml tree.

**Comment by DHowett-MSFT** [comment]
I had a discussion with Austin o*line about this. If we suppress passing `VK_F7` to the Xaml island at the toplevel WndProc, we can feed it directly into the active terminal control. This will require us to take on a bit of technical debt, but perhaps in the name of user NSAT it's worth it. :smile:
